src/move.c

```diff
@@ -1140,7 +1140,14 @@ do_recycle(int descr, dbref player, const char *name)
                         dequeue_prog(thing, 0);
                     }
 
-                    /* FIXME: This is a workaround for bug #201633 */
+                    /*
+                     * @TODO This is a workaround for bug #201633
+                     *
+                     * This bug predates sourceforge and is from a system
+                     * long gone.  MUCKs aren't dying left and right due
+                     * to this, so I'm inclined to remove this code block
+                     * now.
```

Comment:
The workaround might be what causes MUCKs not to die, though? :P  No idea.

Reply:
No, it's not -- the if statement has an assert(0); in it.  MUCKs would be dying left and right if it WAS hitting this code block :)  That's what my comment intends to say here.

Reply:
> No, it's not -- the if statement has an assert(0); in it. MUCKs would be dying left and right if it WAS hitting this code block :) That's what my comment intends to say here.

Oh!  I see.  Thanks for the enlightenment. :D

Reply:
Though, isn't it also possible NDEBUG can sometimes be defined (see autoconf.h.in, config.h) causing assert() to do nothing?

Reply:
NDEBUG causes assert() to be disabled - which is the usual state of the MUCK (--enable-debug I think causes this not to be #defined, I think).

I think I have that logic right.

Reply:
> NDEBUG causes assert() to be disabled - which is the usual state of the MUCK (--enable-debug I think causes this not to be #defined, I think).
> 
> I think I have that logic right.

Correct. I just compiled and confirmed this.

Reply:
Yes, in config.h - Thanks for the confirmation.

Reply:
This was marked as resolved for some reason (which causes this to be nearly hidden on the issue's webpage :c ) even though what we've determined contradicts the conclusion of the comment in question.

Reply:
This was marked as resolved (which causes this to be nearly hidden on the issue's webpage :c ) even though what we've determined contradicts the conclusion of the comment in question.

Reply:
@nyanster, it was a mistake. I unresolved it immediately.

Reply:
@nyanster @wyld-sw So what do we want to do then?  Just leave it as a mystery?  Put a log message in there and see if it shows up anywhere?  The latter would require cooperative MUCKs -- I mean Hope Island would cooperate but we probably need a larger sample size.

Reply:
I would prefer to leave it in (with a log message as you suggest), at least temporarily. I've never seen this message, but until we determine that dequeue_prog always works as expected, I don't think we can remove it.

Reply:
Actually on further thought ... this is SUCH a rare case I'm not sure if we'll see it in the wild.  We'd probably have to test it on a dev MUCK.  How often does a live MUCK @ recycle programs?  I would say hardly ever.

Reply:
> Actually on further thought ... this is SUCH a rare case I'm not sure if we'll see it in the wild. We'd probably have to test it on a dev MUCK. How often does a live MUCK @ recycle programs? I would say hardly ever.

o, true

Reply:
I'm inclined to make this a separate issue, put the issue number in the TODO comment, and try to sort this out on a dev MUCK.  Even then, we may not be able to reproduce it, but we can at least make a concentrated effort to do so and list our test cases out so that if this bug ever does crop up, we've got public history of it rather than a mysterious ticket number in an unknown ticket system.  If that sounds good I'll go ahead and make the ticket for it.

Reply:
> I'm inclined to make this a separate issue, put the issue number in the TODO comment, and try to sort this out on a dev MUCK. Even then, we may not be able to reproduce it, but we can at least make a concentrated effort to do so and list our test cases out so that if this bug ever does crop up, we've got public history of it rather than a mysterious ticket number in an unknown ticket system. If that sounds good I'll go ahead and make the ticket for it.

Sounds good to me.

Reply:
It does sound good, thanks!